Project-Descriptions-and-Plans/LLVM-Toolchain/llvm-toolchain-pl.md

```diff
@@ -0,0 +1,314 @@
+
+In work done a few years ago, code generated for Embench tests by LLVM 11.0 was 4% quicker that GCC 10.2 code. This is an additional reason to offer LLVM support for CORE-V processors.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+
+The "original" compiler used for the development of RISC-V cores is GCC, licensed under the GNU General Public License, requiring vendors to make source code available.  LLVM offers a "permissive" licensing scheme which some customers find more attractive, since it does not require the source to be made available. As a result LLVM-based compiler chains appear to be at the basis (in whole or in part) of a number of proprietary compilers.
+
+The downside is that there is less pressure on companies to contribute back to LLVM development.  This project will give OpenHW Group members a choice over which approach to adopt (make their toolchain code available or not).
+
+### Related work
+
+The upstream LLVM tool chain project already supports standard RISC-V, including all ratified ISA extensions.  RISC-V support in LLVM is strongly compliant to the RISC-V specification, but relatively immature in comparison to older architectures as x86, Arm and MIPS.
+
+Moreover, Clang/LLVM follows closely new ISA extension discussions as it already provides a mechanism to support yet to be ratified extensions.
+
+Other toolchains available for RISC-V are:
+
+- the GNU tool chain for RISC-V, for which a CORE-V specific project is in progress within OpenHW Group;
+- IAR RISC-V compiler
+
+The GNU tool chain project within OpenHW Group mirrors this project in its suport for CORE-V archhitecture.
+
+An OpenHW Group architecture project devoted to the CVA6 (Ariane) family of cores has passed the project concept gate.  There is a clear synergy between this project and the CVA6 project:
+
+- CVA6 hardware design and verification will directly benefit from the LLVM work;
+- in return, the CVA6 project will directly provide inputs (specifications, documentation) for the compiler- and simulator-related activities in the present project.
+
+The various CVE projects will similarly take benefit from LLVM development.
+
+## OpenHW Members/Participants committed to participate
+
+## Project Leader(s)
+### Technical Project Leader(s)
+
+Charlie Keaney, Embecosm
+
+### Project Manager, if a PM is designated
+
+None required.
+
+## Project Documents
+### Project Planning Documents
+
+None required.
+
+### Project Output Documents
+
+None required.
```

Comment:
We will complete the project with no documents? Do we add something to the upstream documentation? Do we not want to add something to the OpenHW technical documentation suite on readthedocs to refer to this work as being available?

Reply:
Well documentation of all types is integral to the upstream development process. I will explain this here.  All the documents beyond this come under the marketing heading to tell people what is available.